### PR TITLE
improve: pscanrules: CSP rule now uses Salvation v3

### PR DESCRIPTION
**Pull request by kingthorin**
- ContentSecurityPolicyScanRule > Re-write to use new Salvation mechanisms.
- ContentSecurityPolicyScanRuleUnitTest > Updated to assert current behavior.
- pscanrules.html > Updated help content.
- Messages.properties > Remove now unused key value pairs related to multiple CSP handling.
- CHANGELOG > Added change note.

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

File: addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/ContentSecurityPolicyScanRule.java [hunk 1]
```diff
@@ -313,54 +302,89 @@ private String getCSPNoticesString(ArrayList<Notice> notices) {
...
-        if (pol.allowsPrefetchFromSource(PARSED_WILDCARD_URI)) {
-            allowedSources.add("prefetch-src");
-        }
-        if (pol.allowsFormAction(PARSED_WILDCARD_URI)) {
-            allowedSources.add("form-action");
+        Policy pol = Policy.parseSerializedCSP(policyText, PolicyErrorConsumer.ignored);
+        Optional<URI> httpUri = URI.parseURI("http://" + RAND_FQDN);
+        Optional<URI> httpsUri = URI.parseURI("https://" + RAND_FQDN);
+
+        if (httpUri.isPresent() && httpsUri.isPresent()) {
```
Comment (kingthorin) [inline addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/ContentSecurityPolicyScanRule.java:312]: SonarLint insisted these be checked. I think it's kinda over-kill given that the value is being set directly/specifically, but maybe I'm missing some edge case about `Optional`s

**Comment by kingthorin** [comment]
Fixed

**Comment by thc202** [comment]
The commit/PR title should be updated as well. I didn't take a closer look to the code changes yet.

**Comment by kingthorin** [comment]
Sure I’ll tweak them too. The issue is that they’ve called it Salvation 2 for release/Maven but it is actually version 3 😞

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]

**Pull request merged**